src/modeler/AutoRest.Swagger/Validation/NonAppJsonTypeWarning.cs

```diff
@@ -30,6 +30,11 @@ public override bool IsValid(string entity)
             => (entity.ToLowerInvariant().StartsWith(AppJsonType));
         
         /// <summary>
+        /// What kind of change implementing this rule can cause.
+        /// </summary>
+        public override ValidationChangesImpact ValidationChangesImpact => ValidationChangesImpact.ServiceImpactingChanges;
```

Comment:
I'm not sure on the category for this one, if something other than application/json is included, we just report the warning to make sure the service does support it and if not, it gets removed from the swagger, so in terms of impact, it seems like is more of a swagger update other than a service impacting change, on the other hand. It may or may not impact the SDK depending on whether the code generator knows what to do with the other types. Thoughts?